### PR TITLE
fix(router): fallback support in new_defineRouter and new_createPages

### PR DESCRIPTION
**Pull request by dai-shi**
add fallback mechanism for #1003.

### CODE REVIEW

File: packages/waku/src/router/define-router.ts [hunk 1]
```diff
@@ -302,12 +302,14 @@ export function new_defineRouter(fns: {
...
     routeElement: ReactNode;
     elements: Record<SlotId, ReactNode>;
+    fallbackElement?: ReactNode;
   }>;
 }): ReturnType<typeof defineEntries> {
   const platformObject = unstable_getPlatformObject();
   type MyPathConfig = {
     pattern: string;
     pathname: PathSpec;
+    // FIXME Do we need this isStaticRouteElement?
```
Comment (tylersayshi) [inline packages/waku/src/router/define-router.ts:312]: nope! good catch, safe to remove. This just slipped in as left over

**Review by tylersayshi — COMMENTED** [review]
This seems fine. I don't yet fully understand what `fallbackElement` is yet though.

File: packages/waku/src/minimal/client.ts [hunk 2]
```diff
@@ -286,19 +295,32 @@ export const Slot = ({
...
+  const renderSlot = (elements: Record<string, ReactNode>, err?: unknown) => {
     if (!(id in elements)) {
       if (fallback) {
+        if (err) {
+          // HACK I'm not sure if this is the right way
+          return createElement(
+            ChildrenContextProvider,
+            { value: createElement(InnerErr, { err }) },
+            fallback,
+          );
```
Comment (dai-shi) [inline packages/waku/src/minimal/client.ts:322]: I wonder if this hack works for new_createPages fallback. Let check it after merging...

File: packages/waku/src/minimal/client.ts [hunk 3]
```diff
@@ -309,11 +331,17 @@ export const Slot = ({
       elements[id],
     );
   };
+  if (unstable_renderPrev) {
+    if (!elementsPromise.prev) {
+      throw new Error('Missing prev elements');
+    }
+    return renderSlot(elementsPromise.prev);
+  }
```
Comment (dai-shi) [inline packages/waku/src/minimal/client.ts:339]: I'm not happy with this, and hope to redesign it after finishing our current transition.

File: packages/waku/src/minimal/client.ts (hunk 2 above)
Comment (dai-shi) [reply packages/waku/src/minimal/client.ts:322]: It doesn't work. so #1015.

**Pull request merged**